### PR TITLE
validate value length

### PR DESCRIPTION
**Pull request by snopoke**
Pulled from: https://github.com/dimagi/commcare-hq/pull/15537

Pending disucussion:
https://docs.google.com/document/d/1jl0ujQyYGqEvcu39E67pGmRGDMfoPTvqoCngMv37Cfs/edit#

buddy @mkangia

### CODE REVIEW

File: corehq/form_processor/tests/test_basic_cases.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from django.test import TestCase
 from django.test.utils import override_settings
 
+from casexml.apps.case.exceptions import CaseValueError
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_basic_cases.py]: F401 'CaseValueError' imported but unused

**Comment by benrudolph** [comment]
this is a tough question, @ctsims may have some insights. left some comments/questions on the doc

**Comment by mkangia** [comment]
@snopoke i think i will keep out of this one and let others weigh in for review.

**Comment by snopoke** [comment]
:white_check_mark:

**Review by benrudolph — APPROVED** [review]
looks good

**Pull request merged**